File: packages/ios/Sources/CivUI/CivDownloadLink.swift
// CivUI — CivDownloadLink for SwiftUI
// Accessible download link following government design system patterns.

import SwiftUI

/// Accessible download link for government applications.
///
/// Renders a link that triggers a file download.
/// Mirrors the web `civ-download-link` component.
///
/// Usage:
/// ```swift
/// CivDownloadLink(
///     label: "Download form",
///     href: "https://example.gov/form.pdf",
///     filename: "VA-Form-21.pdf",
///     fileSize: "2.4 MB"
/// )
/// ```
public struct CivDownloadLink: View {
    // MARK: - Properties

    /// Link text.
    public var label: String

    /// URL of the file to download.
    public var href: String

    /// Suggested filename for the download.
    public var filename: String

    /// Human-readable file size (e.g., "2.4 MB").
    public var fileSize: String

    /// Whether the link is disabled.
    public var isDisabled: Bool

    // MARK: - Internal State

    @Environment(\.colorScheme) private var colorScheme

    // MARK: - Initializer

    public init(
        label: String = "",
        href: String = "",
        filename: String = "",
        fileSize: String = "",
        isDisabled: Bool = false
    ) {
        self.label = label
        self.href = href
        self.filename = filename
        self.fileSize = fileSize
        self.isDisabled = isDisabled
    }

    // MARK: - Body

    public var body: some View {
        EmptyView()
    }

    // MARK: - Color Helper

    private func adaptiveColor(light: Color, dark: Color) -> Color {
        colorScheme == .dark ? dark : light
    }
}

// MARK: - Preview

#if DEBUG
struct CivDownloadLink_Previews: PreviewProvider {
    static var previews: some View {
        CivDownloadLink(
            label: "Download form",
            href: "https://example.gov/form.pdf",
            filename: "VA-Form-21.pdf",
            fileSize: "2.4 MB"
        )
        .padding()
    }
}
#endif

File: packages/navigation/src/download-link/civ-download-link.ts
import { html, nothing } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { CivBaseElement, LightDomTextMixin } from '@civui/core';
import '../link/civ-link.js';

/**
 * CivUI Download Link
 *
 * A link that triggers a file download. Renders a download icon and
 * optionally displays the file size.
 *
 * @element civ-download-link
 *
 * @prop {string} href - URL of the file to download
 * @prop {string} label - Link text
 * @prop {string} filename - Suggested filename for the download
 * @prop {string} fileSize - Display file size (e.g., "2.4 MB")
 *
 * @example
 * ```html
 * <civ-download-link href="/forms/10-10EZ.pdf" label="Download VA Form 10-10EZ"
 *   filename="10-10EZ.pdf" file-size="1.2 MB">
 * </civ-download-link>
 * ```
 */
@customElement('civ-download-link')
export class CivDownloadLink extends LightDomTextMixin(CivBaseElement) {
  @property({ type: String }) label = '';
  @property({ type: String }) href = '';
  @property({ type: String }) filename = '';
  @property({ type: String, attribute: 'file-size' }) fileSize = '';
  @property({ type: Boolean, reflect: true }) disabled = false;

  override render() {
    const text = this.label || this._initialText;

    return html`
      <civ-link
        label="${text}"
        href="${this.href}"
        download="${this.filename || ''}"
        icon-start="download"
        ?disabled="${this.disabled}"
      ></civ-link>${this.fileSize ? html`
      <span class="civ-text-sm civ-text-base civ-ms-1">(${this.fileSize})</span>` : nothing}
    `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'civ-download-link': CivDownloadLink;
  }
}
